main.nf

```diff
@@ -439,8 +439,9 @@ process bwa_mapped {
     """
     for i in $input_files
     do
-      printf "\${i}\t"
-      samtools idxstats \${i} | awk '{s+=\$3}END{print s}'
+      mapped="\$(samtools idxstats \${i} | awk '{s+=\$3}END{print s}')"
+      unmapped="\$(samtools idxstats \${i} | awk '{s+=\$4}END{print s}')"
```

Comment:
You're running `samtools` twice here. It should be possible to just use `awk` to get the two columns we want in a single go and do everything in one line. Something like:

```bash
samtools idxstats \${i} | awk -v i="\$i" 'BEGIN { mapped += \$3, unmapped += \$4 } END {print \$i, "\t", \$mapped, "\t", \$unmapped, "\n"}'
```

* _caveat:_ I'm not very experienced using `awk` and this could be completely wrong...